### PR TITLE
logging: ensure that MagicArgv1 is always argv1

### PR DESCRIPTION
**Pull request by Zheaoli**
fix: #1054 

Signed-off-by: Zheao.Li <me@manjusaka.me>

### CODE REVIEW

File: pkg/strutil/strutil_test.go [hunk 1]
```diff
@@ -281,3 +281,11 @@ func TestParseCSVMap(t *testing.T) {
...
 	}
 }
+
+func TestOrderedQuery(t *testing.T) {
+	query := NewOrderedQuery()
+	query.Set("abc", "1")
+	query.Set("aa", "2")
+	result := query.Encode()
+	assert.Equal(t, result, "abc=1&aa=2", "")
+}
```
Comment (AkihiroSuda) [inline pkg/strutil/strutil_test.go:291]: Sorry, this is not correct, as the ordering is ignored on parsing back the string to `*net.URL`


https://github.com/containerd/containerd/blob/e85b5a0b81a610f564c3240c69d8f7324b5ecadb/pkg/process/io_util.go#L31

Comment (Zheaoli) [reply pkg/strutil/strutil_test.go:291]: May I ask if it is correct if we sort the args when in the nerdctl xmain function?

**Comment by AkihiroSuda** [comment]
Opened an alternative PR https://github.com/containerd/nerdctl/pull/1056

File: pkg/strutil/strutil_test.go (hunk 1 above)
Comment (AkihiroSuda) [reply pkg/strutil/strutil_test.go:291]: > May I ask if it is correct if we sort the args when in the nerdctl xmain function?

No, that doesn't work with `nerdctl run busybox echo "_NERDCTL_INTERNAL_LOGGING"`


>  Maybe we don't need to concern about the args order?

Need

**Pull request closed without merging**